pycbc/events/significance.py

```diff
@@ -187,6 +187,19 @@ def get_n_louder(back_stat, fore_stat, dec_facs,
     Wrapper to find the correct n_louder calculation method using standard
     inputs
     """
+    # This nan to inf conversion is a safety shield; there shouldn't be any
+    # nan statistic values, but if there are, they should be considered as
+    # quiet as possible. nans are considered greater than any floats in an
+    # argsort
+    nanmask = np.isnan(back_stat)
+    if any(nanmask):
+        logging.warning(
+            "Removing %d NaN background statistic values",
+            np.count_nonzero(nanmask),
+        )
+        back_stat = copy.deepcopy(back_stat)
+        back_stat[nanmask] = -np.inf
```

Comment:
seems like we want something like `back_stat_nonan = numpy.where(nanmask, -np.inf, back_stat)` ?